getting_started/workflow/best_practices/what_are_godot_classes.rst

```diff
@@ -0,0 +1,137 @@
+
+What are Godot classes really?
+==============================
+
+Godot offers two main means of creating types: scripts and scenes.
+Both of these represent a "class" since Godot revolves around
+Object-Oriented design. *How* they do this may not be clear to beginner
+or intermediate users though.
+
+Godot Engine provides classes out-of-the-box (like
+:ref:`Node <class_Node>`), but user-created types are not actually classes.
+Instead they are resources that tell the engine a sequence of initializations
+to perform on an engine class.
+
+Godot's internal classes have methods that register a class's data with
+a :ref:`ClassDB <class_ClassDB>`. This database provides runtime access to
+class information (also called "reflection"). Things stored in the ClassDB
+include, among other things...
+
+- properties
+
+- methods
+
+- constants
+
+- signals
+
+Furthermore, this ClassDB is what Objects actually check against when
+performing any operation. Access a property? Call a method? Emit a signal?
+It will check the database's records (and the records of the Object's base
+types) to see if the Object supports the operation. Every C++ Object defines
+a static `_bind_methods()` function that describes what C++ content it
+registers to the database and how.
+
+So, if the engine provides all of this data at startup, then how does
+a user define their own data? It'd be nice if users could define a custom
+set of data to be appended to an object's data. That way, users could inject
+their own properties and methods into the engine's Object query requests.
+
+*This* is what a :ref:`Script <class_Script>` is. Objects check their attached
+script before the database, so scripts can even override methods.
+If a script defines a `_get_property_list()` method, that data is appended to
+the list of properties the Object fetches from the ClassDB. The same holds
+true for other declarative code.
+
+This can lead to some users' confusion when they see a script as being
+a class unto itself. In reality, the engine just auto-instantiates the
+base engine class and then adds the script to that object. This then allows
+the Object to defer to the Script's content where the engine logic deems
+appropriate.
+
+A problem does present itself though. As the size of Objects increases,
+the scripts' necessary size to create them grows much, much larger.
+Creating node hierarchies demonstrates this. Each individual Node's logic
+could be several hundred lines of code in length.
+
+let's see a simple example of creating a single Node as a child.
+
+.. tabs::
+  .. code-tab:: gdscript GDScript
+    
+    # main.gd
+    extends Node
+    
+    var child # define a variable to store a reference to the child
+
+    func _init():
+        child = Node.new() # Construct the child.
+        child.name = "Child" # Change its name.
+        child.script = preload("child.gd") # Give it custom features.
+        child.owner = self # Serialize this node if self is saved.
```

Comment:
I'm not sure if this should really be here since, if one serialized the node with a child, then instantiated the scene, they'd re-execute the _init of the script and add *another* child in addition to deserializing the previously serialized child. Right?